composer/loggers/mosaic_logger.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log to MPlatform."""
+
+from __future__ import annotations
+
+import logging
+import os
+import time
+from typing import Any, Dict, Optional
+
+from composer.loggers.logger import format_log_data_value
+from composer.loggers.logger_destination import LoggerDestination
+from composer.utils import MissingConditionalImportError, dist
+
+log = logging.getLogger(__name__)
+
+__all__ = ['MosaicLogger']
+
+
+class MosaicLogger(LoggerDestination):
+    """Log to MPlatform.
+
+    Logs metrics to MPlatform. Logging only happens on rank 0 every ``log_interval`` seconds to
+    avoid performance issues.
+
+    Args:
+        log_interval (int, optional): Buffer log calls more frequent than ``log_interval`` seconds.
+                                      Defaults to 60.
+    """
+
+    def __init__(
+        self,
+        log_interval: int = 60,
+    ) -> None:
+        self.log_interval = log_interval
+        self._enabled = dist.get_global_rank() == 0
+        if self._enabled:
+            self.time_last_logged = 0
+            self.buffered_metadata: Dict[str, Any] = {}
+
+            try:
+                import mcli
+                del mcli
+            except ImportError as e:
+                raise MissingConditionalImportError(extra_deps_group='mcli',
+                                                    conda_package='mcli',
+                                                    conda_channel='conda-forge') from e
+            self.run_name = os.environ.get('RUN_NAME')
```

Comment:
I'll add it to the top. Going to skip constants file because we don't have many constants, and it's easier to read everything in a single self contained file, especially if users want to copy paste and add a variant callback